src/app/infokit/infokit.pipe.ts

```diff
@@ -0,0 +1,51 @@
+import { Pipe, PipeTransform} from '@angular/core';
+
+@Pipe ({
+  name: 'infokitPipe'
+})
+
+export class InfokitPipe implements PipeTransform {
+  transform(dict: any[], args: any= []) {
+    let activity: any[] = [];
+    for (let key in dict) {
+      if (dict.hasOwnProperty(key)) {
+        switch (key) {
+          case 'malaria_def' : {
```

Comment:
this is VERY redundant, especially since you already are using a dictionary.

everything in the `switch` can be summarized as:
```
activity.push({key: key, def: dict[key].def, val: dict[key].val});
```

if `dict` has the following structure:
```
{
    malaria_def: {
        def: 'Malaria',
        val: 'values...'
    }
}
```

and this is just one approach